android/app/proguard-rules.pro

```diff
@@ -48,6 +48,15 @@
 -keepclassmembers class *  { @com.facebook.react.uimanager.annotations.ReactProp <methods>; }
 -keepclassmembers class *  { @com.facebook.react.uimanager.annotations.ReactPropGroup <methods>; }
 
+-keep class com.facebook.react.bridge.ModuleRegistryHolder { *; }
```

Comment:
Note that before this PR proguard wasn't enabled for any of the builds. So I guess it will work before this PR even if you delete all the rules. At least this is my understanding, please correct me if I'm wrong. This PR adds `minifyEnabled true` which enables proguard and then you start seeing the errors.

I don't have solid reasoning behind every rule that I have added. As I mentioned in previous comments I started testing to discover the classes that are missing after enabling minify and I added exception for every error that I have discovered.